packages/nc-gui/components/smartsheet/column/BarcodeOptions.vue

```diff
@@ -0,0 +1,99 @@
+<script setup lang="ts">
+import type { UITypes } from 'nocodb-sdk'
+import { AllowedColumnTypesForQrAndBarcodes } from 'nocodb-sdk'
+import type { SelectProps } from 'ant-design-vue'
+import { useVModel } from '#imports'
+
+const props = defineProps<{
+  modelValue: any
+}>()
+
+const emit = defineEmits(['update:modelValue'])
+
+const meta = inject(MetaInj, ref())
+
+const activeView = inject(ActiveViewInj, ref())
+
+const reloadDataHook = inject(ReloadViewDataHookInj)!
+
+const { fields, metaColumnById } = useViewColumns(activeView, meta, () => reloadDataHook.trigger())
+
+const vModel = useVModel(props, 'modelValue', emit)
+
+const { setAdditionalValidations, validateInfos, column } = useColumnCreateStoreOrThrow()
+
+const columnsAllowedAsQrValue = computed<SelectProps['options']>(() => {
+  return fields.value
+    ?.filter(
+      (el) =>
+        el.fk_column_id && AllowedColumnTypesForQrAndBarcodes.includes(metaColumnById.value[el.fk_column_id].uidt as UITypes),
+    )
+    .map((field) => {
+      return {
+        value: field.fk_column_id,
+        label: field.title,
+      }
+    })
+})
+
+const supportedBarcodeFormats = [
+  { value: 'CODE128', label: 'CODE128' },
+  { value: 'upc', label: 'UPC' },
+  { value: 'EAN13', label: 'EAN-13' },
+  { value: 'EAN8', label: 'EAN-8' },
+  { value: 'EAN5', label: 'EAN-5' },
+  { value: 'EAN2', label: 'EAN-2' },
+  { value: 'CODE39', label: 'CODE39' },
+  { value: 'ITF14', label: 'ITF-14' },
+  { value: 'MSI', label: 'MSI' },
+  { value: 'PHARMACODE', label: 'pharmacode' },
+  { value: 'CODABAR', label: 'codabar' },
+]
+
+onMounted(() => {
+  // set default value
+  vModel.value.fk_barcode_value_column_id = (column?.value?.colOptions as Record<string, any>)?.fk_barcode_value_column_id || ''
+})
+
+setAdditionalValidations({
+  fk_barcode_value_column_id: [{ required: true, message: 'Required' }],
+  barcode_format: [{ required: true, message: 'Required' }],
+})
```

Comment:
Need a better message

<img width="471" alt="image" src="https://user-images.githubusercontent.com/35857179/208658460-1618c19e-955c-41a8-b425-e0ba45063e60.png">

Reply:
I preset the field so the problem arises now only when there's not appropriate field yet. Here how it looks like: 


<img width="401" alt="Screenshot 2022-12-21 at 13 04 35" src="https://user-images.githubusercontent.com/35769846/208901585-260d1686-7d9e-40b3-93ef-4f87bdece5c4.png">

Do you think it's correct way to resolve this issue?

Reply:
What about adding a small question mark icon next to `Column with Barcode value` as a help text? Like hovering would show the available column types for barcode type. In this way, we can keep the no-data-text short (we can say no valid column type, please refer to the help text blah blah blah) and at the same time we can still allow users to check that even if there are some available columns.

Reply:
Yey! that's sound very good. Thank you!

Reply:
I've changed it, can you check it?

Reply:
It looks good!